### ISSUE TITLE
Problems in test_generic_methods

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#4336]
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/base/tests/test_generic_methods.py#L54

`table1` is defined but never used.  Any idea what the intention is there?

https://github.com/statsmodels/statsmodels/blob/master/statsmodels/base/tests/test_generic_methods.py#L246

```
        exog = sm.add_constant(data.exog, prepend=False)
        mod = sm.NegativeBinomial(data.endog, data.exog)
```

`exog` is edited but then unused, while `data.exog` is passed.  If instead `exog` is passed, 4 tests fail.

**Comment by josef-pkt** [comment]
table1 might just be missing the assert
assert_allclose(table1, table_res, rtol=1e-12)

the exog definition is either redundant, if data.exog includes already  a constant, or it's a case that didn't work nicely at the time.

**Comment by jbrockmendel** [comment]
> table1 might just be missing the assert

Looking at it more closely, I think table1 was replaced with table2 in the next line, and table 1 was just never deleted.

> the exog definition is either redundant, if data.exog includes already a constant, or it's a case that didn't work nicely at the time.

I'm not sure how to make this actionable.  Suggestions?  (obviously not actionable _here_, but I'm trying to actually fix these downstream)

**Comment by josef-pkt** [comment]
> table1 might just be missing the assert
Looking at it more closely, I think table1 was replaced with table2 in the next line, and table 1 was just never deleted.

These tests are consistency check, so table1 should also be checked, unless it is already done in another assert.

> I'm not sure how to make this actionable. Suggestions? (obviously not actionable here, but I'm trying to actually fix these downstream)

Check whether data.exog includes a constant. 
If yes, delete the unused exog definition.
If data.exog does not include a constant, then we should check why using the defined `exog` doesn't or didn't work. 
(I/we did quite a bit of work on NegativeBinomial since those unit tests were written.
